src/validation/body-check.ts

```diff
@@ -0,0 +1,32 @@
+import { Context } from 'hono';
+import { z, ZodError } from 'zod';
+
+export const BodySchema = z.object({
+  id: z.number().optional(),
+  name: z
+    .string()
+    .min(1)
+    .max(50)
+    .regex(/^[A-Za-z ]+$/),
```

Comment:
1. Will it work? `/^[a-z\s]+$/i`
2. Please use `\s` in regex instead an actual space ` `